_posts/2022-12-06-integrating-auth0-with-react.md

```diff
@@ -0,0 +1,317 @@
+
+Beautiful, Users can now Sign In, Sign Up and perform Password recoveries all with this universal login page.
+
+### Great, Now how about logout?
+
+Ready to see them go already?, Well this is just as easy.
+
+Firstly let's create a new Logout button.
+
+```tsx
+// /src/components/LogoutButton.tsx
+
+import React, { FC } from 'react';
+import { useAuth0 } from "@auth0/auth0-react";
+
+const LogoutButton: FC = () => {
+    const { logout } = useAuth0();
+
+    const style = {width: '200px', height: '50px', margin: '20px'} // Give the button a bit of substance
+
+    return <button style={style} onClick={() => logout()}>Logout</button>;
+}
+
+export default LogoutButton;
+```
+
+Here we will once again utilise the `useAuth0` hook this time with the `logout` method, Pretty straight forward right?
+
+The `logout` method will direct your users to your Auth0 logout endpoint, and once completed, will redirect them to your root page (i.e. `http://localhost:3000`).
+
+It can also be configured with an optional parameter `returnTo` to direct them to another part of your application if you would like.
+```ts
+logout({ returnTo: 'http://localhost:3000/logout' })
+```
+
+**Be aware, this returnTo parameter MUST be included in your Allowed Logout URLs that we configured earlier**
+
+```tsx
+// /src/App.tsx
+...
+import LoginButton from './components/LoginButton';
+import LogoutButton from './components/LogoutButton';
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        ...
+        <a
+          ...
+        >
+          Learn React
+        </a>
+        <LoginButton />
+        <LogoutButton />
+      </header>
+    </div>
+  );
+}
+...
+```
+
+![Logout Button](/img/blog/integrating-auth0-with-react/logout.png){:width="275px" height="190px" loading="lazy"}
+
+Awesome, we now have User Signup, Login, and Logout!
+
+## User information and Profile
+
+Now that our users can login, How do we go about getting their information such as Name, Email address, Profile Picture etc?
+
+Alright, Let's create a new component that will retrieve that for us.
+
+```tsx
+// /src/components/Profile.tsx
+
+import React, { FC } from 'react';
+import { useAuth0 } from "@auth0/auth0-react";
+
+const Profile: FC = () => {
+    const { user, isAuthenticated, isLoading } = useAuth0();
+
+    if (isLoading) {
+        return <div>Loading...</div>
+    }
+
+    return (
+        <div>
+            {isAuthenticated && user && (
+                <>
+                    <img src={user.picture} alt={user.name} />
+                    <h2>{user.name}</h2>
+                    <p>{user.email}</p>
+                </>
+            )}
+        </div>
+    );
+}
+
+export default Profile;
+```
+
+Once again, the `useAuth0` hook showing its strengths! We can get the user object, check if they are authenticated, and that everything has been fully loaded before displaying the user profile.
+
+Let's add that to our homepage
```

Comment:
perhaps another word other than Let's

Reply:
Good call, I overused that a bit at the end there